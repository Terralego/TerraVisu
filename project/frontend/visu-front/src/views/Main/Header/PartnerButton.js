import React, { useState } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';
import {
  Overlay,
  Classes,
} from '@blueprintjs/core';

import withDeviceSize from '@terralego/core/hoc/withDeviceSize';

import NavBarItemDesktop from '@terralego/core/components/NavBarItemDesktop';
import NavBarItemTablet from '@terralego/core/components/NavBarItemTablet';

import PartnerPage from './PartnerPage';

export const PartnerButton = ({ isMobileSized, isPhoneSized, ...props }) => {
  const [isOpen, setOpen] = useState(false);
  const NavBarItem = isMobileSized ? NavBarItemTablet : NavBarItemDesktop;

  return (
    <>
      <NavBarItem {...props} onClick={() => setOpen(true)} />
      <Overlay
        className={classNames(
          Classes.OVERLAY_SCROLL_CONTAINER,
          Classes.LIGHT,
          'modal-mentions-legales',
        )}
        isOpen={isOpen}
        onClose={() => setOpen(false)}
      >
        <div
          className={classNames(
            Classes.CARD,
            Classes.ELEVATION_4,
          )}
        >
          <PartnerPage />
        </div>
      </Overlay>
    </>
  );
};

PartnerButton.propTypes = {
  isMobileSized: PropTypes.bool,
  isPhoneSized: PropTypes.bool,
};

PartnerButton.defaultProps = {
  isMobileSized: false,
  isPhoneSized: false,
};


export default withDeviceSize()(PartnerButton);
