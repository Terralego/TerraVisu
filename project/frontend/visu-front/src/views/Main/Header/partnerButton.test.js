import React from 'react';
import renderer from 'react-test-renderer';
import { PartnerButton } from './PartnerButton';

jest.mock('@terralego/core/components/NavBarItemTablet', () => () => <div className="container">NavBarItemTablet</div>);
jest.mock('@terralego/core/components/NavBarItemDesktop', () => () => <div className="container">NavBarItemDesktop</div>);

it('should render correctly in mode desktop', () => {
  const tree = renderer.create(
    <PartnerButton />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly in mode mobile', () => {
  const tree = renderer.create(
    <PartnerButton isMobileSized />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
