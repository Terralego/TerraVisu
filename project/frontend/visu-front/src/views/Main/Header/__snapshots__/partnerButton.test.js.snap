// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly in mode desktop 1`] = `
<div
  className="container"
>
  NavBarItemDesktop
</div>
`;

exports[`should render correctly in mode mobile 1`] = `
<div
  className="container"
>
  NavBarItemTablet
</div>
`;
